sql/schema/core.sql

```diff
@@ -0,0 +1,86 @@
+-- enable UUID of PostgreSQL
+CREATE EXTENSION IF NOT EXISTS pgcrypto;
+
+--users information 
+create table users(
+    id  UUID primary key default gen_random_uuid(),
+    username VARCHAR(255) NOT NULL UNIQUE,
+    email VARCHAR(255) NOT NULL UNIQUE,
+    password_hash VARCHAR(255) NOT NULL,
+    deleted_at TIMESTAMPTZ NULL--softdelete
+);
+-- index
+CREATE INDEX idx_users_username ON users(username);
+CREATE INDEX idx_users_email ON users(email);
+CREATE INDEX idx_users_deleted_at ON users(deleted_at) WHERE deleted_at IS NOT NULL;
```

Comment:
```suggestion
CREATE INDEX idx_users_email ON users(email);
```
基本的にemailくらいしかindexをはる必要を感じないのですが、usename, deteled_atにindexを貼り使用されるusecseはどのようなものを想定してますか？